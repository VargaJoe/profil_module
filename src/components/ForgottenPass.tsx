
import * as React				    from 'react';
import axios                        from 'axios';

const DATA = require('../config.json');

export class ForgottenPassComponent extends React.Component<any, any> {
    constructor(props: any) {
        super(props);
        this.state = {
            email: ''
        },
        this.clickHandler           = this.clickHandler.bind(this);
        this.handleChange           = this.handleChange.bind(this);
    }

    public clickHandler = () => {
        const userEmail = {
            UserEmail:  this.state.email
        };
        axios.post(DATA.odataDomain + '/Root/Sites/Profil(\'ForgottenPassword\')/BisonProfileSendChangePasswordMail', userEmail)
        .then(res => {
            this.props.history.push('/login');
        });
    }

    public handleChange = (e: any) => {
        this.setState({
            email: e.target.value,
        });
    }

    // edgar.dovbak@sensenet.com

    render() {

		return (
			<div className="">
				<label htmlFor="Email">Enter your email</label>
                <input 
                    type="email" 
                    id="Email"
                    name="Email"
                    autoComplete="nope"
                    required={true}
                    onChange={this.handleChange}
                />
                <p/>
                <button className="sn_btn" onClick={this.clickHandler} >
                    Send
                </button>
			</div>
		);
	}
}

export default ForgottenPassComponent;
 