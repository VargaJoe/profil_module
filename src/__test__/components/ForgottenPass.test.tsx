import * as React 						  	from 'react';
import ForgottenPass,
		{ ForgottenPassComponent }			from '../../components/ForgottenPass';
import { 
	configure, 
		shallow,
		mount } 							from 'enzyme';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<ForgottenPass /> shallow rendering', () => {
		
	const forgottenPass = mount(<ForgottenPass />); 
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(forgottenPass)).toMatchSnapshot();
	});

	it('Test handleChange', () => {
		const fff = forgottenPass;
		const spy = jest.spyOn((fff.instance() as any), 'handleChange');
		let e = {
			target: {
				value: 'email@test.test'
			}
		};
		const ddd = (fff.instance() as any).handleChange(e);
		expect(spy).toHaveBeenCalledTimes(1);
	});

	it('Test clickHandler ', () => {
		const fff = forgottenPass;
		const spy = jest.spyOn((fff.instance() as any), 'clickHandler');
		const ddd = (fff.instance() as any).clickHandler();
		expect(spy).toHaveBeenCalledTimes(1);
	});
});
 