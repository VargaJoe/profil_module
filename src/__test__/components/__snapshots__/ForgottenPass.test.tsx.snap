// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgottenPass /> shallow rendering Match to snapshot 1`] = `
<ForgottenPassComponent>
  <div
    className=""
  >
    <label
      htmlFor="Email"
    >
      Enter your email
    </label>
    <input
      autoComplete="nope"
      id="Email"
      name="Email"
      onChange={[Function]}
      required={true}
      type="email"
    />
    <p />
    <button
      className="sn_btn"
      onClick={[Function]}
    >
      Send
    </button>
  </div>
</ForgottenPassComponent>
`;
